keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/constraints/Ec2CanaryConstraintDeployHandler.kt

```diff
@@ -54,17 +54,16 @@ class Ec2CanaryConstraintDeployHandler(
     val scope = CoroutineScope(GlobalScope.coroutineContext)
     val judge = "canary:${deliveryConfig.application}:${targetEnvironment.name}:${constraint.canaryConfigId}"
 
-    val image = imageService.getLatestNamedImageWithAllRegionsForAppVersion(
-      // TODO: Frigga and Rocket version parsing are not aligned. We should consolidate.
-      appVersion = try {
-        AppVersion.parseName(version.replace("~", "_"))
-      } catch (ex: Exception) {
-        throw SystemException("trying to parse name for version $version but got an exception", ex)
-      },
+    val images = imageService.getLatestNamedImageForAppVersionInRegions(
+      appVersion = AppVersion.parseName(version.replace("~", "_")),
```

Comment:
Any particular reason why you removed the `try/catch` block here?

Reply:
I actually meant to go back and make a standard extension function for this. It appears like there are 2 error modes, one in which it throws an exception and one in which it returns `null`.